### ISSUE TITLE
How to reshape an pre trained data input. 

### ISSUE THREAD

**Issue by xhsoldier** [facebookarchive/caffe2#614]
Like in caffe one, I can reshape the data layer:
` net.blobs['data'].reshape(1,3,ws,hs)`

In caffe2, I create the net using the pretrained model like this:
```
INIT_NET = 'net_init.pb'
PREDICT_NET = 'net_predict.pb' 
init_net = _parseFile(INIT_NET)                                                             
predict_net = _parseFile(PREDICT_NET)
predict_net.name = 'predict_net'                                                                
workspace.RunNetOnce(init_net)                                                                  
workspace.CreateNet(predict_net)  
net = workspace.Predictor(init_net_12.SerializeToString(), predict_net_12.SerializeToString()) 

#this is wrong
workspace.FetchBlob('data').reshape(1,3,ws,hs)
result = net.run([scale_img])
```
 **workspace.FetchBlob('data').reshape(1,3,ws,hs)** can not reshape the data input, how to do this?

 workspace.FetchBlob('data').reshape(1,3,ws,hs)   
ValueError: total size of new array must be unchanged

**Comment by andreasmherman** [comment]
Noticed that this issue is still open. Have you made any progress regarding it? I'm stuck on the exact same thing (but I'm running the predictor "GPU-style").

**Comment by xhsoldier** [comment]
No, I am still struggling on this problem. I am trying to porting mtcnn to caffe2, but first I need to reshape the input data.

**Comment by andreasmherman** [comment]
Funny, I'm trying the exact same thing! Have loaded my model in a different way tho, but I can try your way. Gonna try to reproduce your code, what does your `_parseFile` function do?

**Comment by xhsoldier** [comment]
It is caffe2 built-in function, to load the net from the filename.

**Comment by wjxiz1992** [comment]
@xhsoldier hi, I recently  have been searching for caffe2 mtcnn resources and here I see your work. May I know where have you been and what's your progress now? Thanks a lot.

**Comment by xhsoldier** [comment]
I do not have any solutions.

**Comment by alekseic** [comment]
I think that this solution might work
https://github.com/caffe2/caffe2/issues/956

**Comment by jingyugao** [comment]
@wjxiz1992 how is your work now? I am doing the same work now.

**Comment by wjxiz1992** [comment]
@jingyugao Hi, sorry I didn't dig into this problem deeply and I've switched to other topics.
